### PR TITLE
Make Record Name File Timeout Configurable

### PR DESCRIPTION
**Pull request by hmudimi**
[//]: * (If not ready for merging, indicate the reason after the # on the next line.)
#
#### Card(s)
[//]: * (Link to the PivotalTracker or LeanKit cards.)
- [Make Record Name File Timeout Configurable](https://www.pivotaltracker.com/story/show/166847239)

#### Changes Made
1. Updated the VARIABLE type parameter in Record block (UI)
2. Included new CUSTOM type parameter in Record block (UI)
3. Linking toolkit properties with the VARIABLE type parameter
4. Linking toolkit properties with the CUSTOM type parameter

### CODE REVIEW

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java [hunk 1]
```diff
@@ -382,9 +438,36 @@ public void toggleButtonSelected(ToggleButton button) {
...
 			defaultButton.setSelected(false);
+			if(customButton != null)
+				customButton.setSelected(false);
+			if ((flags & VARIABLE) > 0) {
+				variableButton.setSelected(false);
+			}
+			if ((flags & EXPRESSION) > 0) {
+				expressionButton.setSelected(false);
+			}			
+		} else  if(button == customButton){
```
Comment (jcron) [inline com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java]: I see `expressionButton` as an option - is an expression button allowing for the javascript? Is there a reason this couldn't have been an `expressionButton` type with a value of `VirtualHold.getPlatformProperty('com.virtualhold.toolkit.namefilerecordinginitialtimeout')` and not have to add a new button type?

File: com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java [hunk 2]
```diff
@@ -1069,7 +1069,10 @@ protected IDocument renderDataRequest(ILinkFactory links,
 					.booleanValue());
 		}
 		if (initialTimeout != null && initialTimeout.length() > 0) {
+			System.out.println("initialTimeout : " + initialTimeout);
```
Comment (jcron) [inline com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java]: doesn't this get printed in the vxml?

Comment (hmudimi) [reply com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java]: These logs are printing in the Catalina file. VXML generation not getting affected. @jcron

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java (hunk 1 above)
Comment (hmudimi) [reply com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java]: Yes @jcron , We can directly use Expression type button with `VirtualHold.getPlatformProperty('com.virtualhold.toolkit.namefilerecordinginitialtimeout')`. I have already tested this, it is working. I have used VARIABLE type also and tested it had data reading issue which I have fixed. Additionally, we have included a new button named CUSTOM where we can pass variable name dynamically to ValutStack constructor. As per our development plan, I have implemented this and after some analysis, I have understood the working of VARIABLE type and EXPRESSION type and also tested with both options. Some times we are getting NullPointerExceptions which I have fixed already.

1. Do you want me to remove the new CUSTOM button or we can leave it for future development?

As per this development, we don't need CUSTOM, we can use VARIABLE type or EXPRESSION type. We can complete this development with minimal changes by using EXPRESSION.

2. Do you want to change it to EXPRESSION type button?

**Comment by hmudimi** [comment]
@jcron When the client configured initial timeout less than 5seonds, Do we need to set the default value, or is it okay to use configured value?

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java (hunk 1 above)
Comment (jcron) [reply com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java]: Is there any behavior that custom provides that can't be provided by EXPRESSION or VARIABLE? EXPRESSION seems like what we need for this scenario

File: com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java (hunk 2 above)
Comment (jcron) [reply com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java]: I was wondering if this also gets printed in the vxml and if it was necessary to print here

**Comment by jcron** [comment]
> @jcron When the client configured initial timeout less than 5seonds, Do we need to set the default value, or is it okay to use configured value?

it is okay to use configured value - the behavior should be to only use 5 seconds if the configured option is missing. So I would expect that the 5seconds would be in the EXPRESSION

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java (hunk 1 above)
Comment (hmudimi) [reply com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.media.core/src/main/java/org/eclipse/vtp/desktop/media/core/ValueStack.java]: No @jcron  We can achieve the same behavior using EXPRESSION. I will remove the CUSTOM type code and use EXPRESSION.

File: com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java (hunk 2 above)
Comment (hmudimi) [reply com.vht.openvxml.framework/com.vht.openvxml.framework.plugins/org.eclipse.vtp.framework.interactions.voice/src/main/java/org/eclipse/vtp/framework/interactions/voice/services/VoicePlatform.java]: @jcron I have removed the log statement.

**Comment by hmudimi** [comment]
> > @jcron When the client configured initial timeout less than 5seonds, Do we need to set the default value, or is it okay to use configured value?
> 
> it is okay to use configured value - the behavior should be to only use 5 seconds if the configured option is missing. So I would expect that the 5seconds would be in the EXPRESSION

Thanks @jcron 
I will update code as per our discussion and let you know

**Review by jcron — APPROVED** [review]

**Pull request merged**